Topic: Looking for help: Dimensions of profile matrices do not match !

genefisher opened the pull request:
Hey ,

I met some problem in using XC MS, I can not fix by myself now. I am looking for help in here.

I divided my samples(650 samples) into two sub groups named mock and treatment and then put them into one folder to run XCMS.

When I run mock(325 samples) independently, the XC MS runs well and it works fine for me.

When I run treatment(325 samples) independently, the the XC MS also runs well.

However, when I run mock and treatment samples(650 samples) together, the XC MS shows error as follows:

Control0h_L1-50_BD7_01_38715  Create profile matrix with method 'bin' and step 0.1 ... OK
Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-51_BE7_01_38716  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-52_BF7_01_38717  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-53_BG7_01_38718  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-54_BH7_01_38719  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-55_BA8_01_38720  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-56_BB8_01_38721  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-57_BC8_01_38722  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-58_BD8_01_38723  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-59_BE8_01_38724  Create profile matrix with method 'bin' and step 0.1 ... OK
Control0h_L1-6_BG1_01_38666  Create profile matrix with method 'bin' and step 0.1 ... OK
[Found gaps: cut scantime-vector at  1399.01 seconds](url)
Control0h_L1-60_BF8_01_38725  Create profile matrix with method 'bin' and step 0.1 ... OK
### Error in .local(object, ...) :
  Dimensions of profile matrices do not match !
> xset3 <- group(xset2, method="density", mzwid=0.01, sleep=0.001,
+                minfrac=0.1, minsamp=1, bw=5)
Error in group(xset2, method = "density", mzwid = 0.01, sleep = 0.001,  :
  object 'xset2' not found
> xset4 <- fillPeaks(xset3)
Error in fillPeaks(xset3) : object 'xset3' not found
> an <- xsAnnotate(xset4)
Error in xsAnnotate(xset4) : object 'xset4' not found
> anF <- groupFWHM(an, perfwhm = 0.6)
Error in groupFWHM(an, perfwhm = 0.6) : object 'an' not found
> anI <- findIsotopes(anF, mzabs=0.01)
Error in findIsotopes(anF, mzabs = 0.01) : object 'anF' not found
> anIC <- groupCorr(anI, cor_eic_th=0.75)
Error in groupCorr(anI, cor_eic_th = 0.75) : object 'anI' not found
> anFA <- findAdducts(anIC, polarity="positive")
Error in findAdducts(anIC, polarity = "positive") :
  object 'anIC' not found
> write.csv(getPeaklist(anFA), file="data.csv")
Error in getPeaklist(anFA) : object 'anFA' not found



Many thanks for you in advance!

sneumann commented:
Hi, I think something went wrong here, can you close 
and open as issue rather than pull request ? 
Also, please include the output of `sessionInfo()`
Yours, Steffen

jorainer committed:
refactor: multiple smaller changes

- Export `Param` class.
- Don't drop initial feature definitions in the `featureDefinitions<-` method
  allowing to add additional columns to this data frame.

jorainer committed:
feat: filterFile gains parameter keepFeatures

- The `filterFile` method gains parameter `keepFeatures` to force keeping
  correspondence result even due filtering by samples (files).

jorainer committed:
fix: ensure correlate,Chromatogram works even if no points overlap

wkumler committed:
Adding unit tests for XCMSnExp -> xcmsSet method

wkumler committed:
Adding unit test for new XCMSnExp -> xcmsSet method

wkumler committed:
Moving unit tests to correct location

jorainer committed:
feat: add process step for feature grouping

wkumler committed:
Think I found erroneous unit test: ftDef was checking identical to old code not new expected behavior and type should be 'double' rather than 'numeric'

jorainer committed:
feat: add parameter n and value to featureChromatograms

- `featureChromatograms` gets parameters `n` and `value` to extract EICs from
  only the top n samples with highest intensities.
- Add related unit tests and documentation.

jorainer committed:
feat: add normalize method for Chromatogram objects

- Add `normalize,Chromatogram` and `normalize,Chromatograms` to normalize
  intensities in a chromatogram object.
- Add related documentation and unit tests.

jorainer committed:
fix: filterColumnsKeepTop works also on single EIC Chromatograms

jorainer committed:
fix: tentative fix for issue #475

jorainer committed:
ci: check also pushes to jomaster branch

jorainer committed:
Merge pull request #478 from wkumler/master

PR to resolve issue #464 by fixing group matrix type

jorainer committed:
Merge branch 'master' into jomaster

jorainer committed:
ci: force installing MsCoreUtils from github

jorainer committed:
ci: remove travis CI tests

jorainer committed:
fix: conversion from XCMSnExp to xcmsSet keeps phenodata

- Fix issue in conversion from `XCMSnExp` to `xcmsSet` ensuring that a phenodata
  `data.frame` will **always** be set. Thanks to Andris Jankevics for reporting.

jorainer committed:
tests: set random number seed in various unit tests

jorainer committed:
ci: remove testing on macOS devel as dependencies are not available

jorainer committed:
Merge pull request #480 from sneumann/jomaster

Several updates, fixes etc

jorainer committed:
refactor: improve data splitting for parallel peak detection

jorainer committed:
refactor: improve splitting objects for parallel processing

- Improve performance of splitting of objects for paralle processing.
- Should improve the speed of `findChromPeaks`, `refineChromPeaks`,
  `chromPeakSpectra` and `reconstructChromPeakSpectra`.

jorainer committed:
refactor: additional performance improvements

- Further improve splitting an `XCMSnExp` object by file by first splitting the
  chrom peak data.
- Use `extractROWS` instead of `[` for `DataFrame`.

jorainer committed:
fix: minor fixes related to the previous committs

jorainer committed:
fix: tentative fix for issue #481

jorainer committed:
refactor: use bplapply instead of bpmapply in refineChromPeaks

- Small update to refineChromPeaks to use bplapply instead of bpmapply.

jorainer committed:
feat: add fillChromPeaks,ChromPeakAreaParam

- Add `ChromPeakAreaParam` for `fillChromPeaks` that integrates signal from the
  actually detected chromatographic peaks of a feature.
- Add related unit tests and documentation.

jorainer committed:
fix: fix problem in unit test and disable some vignettes

jorainer committed:
Merge pull request #484 from sneumann/jomaster

Performance improvements and new functionality

jorainer committed:
fix: fix error in refineChromPeaks when no peaks found to be merged

jorainer committed:
tests: tentative performance improvement of unit tests

- Avoid repeatedly performing peak detection in examples.
- Slightly tune unit tests.
- Disable parallel processing in vignettes and unit tests.

jorainer committed:
ci: add Bioconductor badges

jorainer committed:
docs: update all examples using the internal test data

jorainer committed:
Moved data from inst/data to data

jorainer committed:
Merge pull request #488 from sneumann/jomaster

New peak filling method

jorainer committed:
fix: report rtime for SWATH reconstructed spectra

- `reconstructChromPeakSpectra` reports the median rt of all MS2 chromatographic
  peaks from which the MS2 spectrum was reconstructed as its retention
  time (issue #485).
- Change default for `expandRt` in `reconstructChromPeakSpectra` to
  `expandRt = 0` to ensure that only signal of identified chromatographic peaks
  is used in the correlation.

jorainer committed:
Merge pull request #489 from sneumann/jomaster

fix: report rtime for SWATH reconstructed spectra

jorainer committed:
refactor: add polarity to reconstructed MS2 spectra (issue #485)

jorainer committed:
refactor: report precursorIntensity in reconstructed spectra

- `reconstructChromPeakSpectra`: report the intensity of the MS1 chromatographic
  peak as the `precursorIntensity` of the reconstructed MS2 spectrum (issue
  #485).

jorainer committed:
Merge pull request #490 from sneumann/jomaster

Some additional fixes to reconstructChromPeakSpectra

jorainer committed:
refactor: use MSpectra and MChromatograms from MSnbase

- Change use to `MSnbase::MSpectra` and `MSnbase::MChromatograms` (issue #491).

jorainer committed:
docs: reduce and tune examples, unit tests and vignettes

- Multiple modifications which aim at lowering the time needed to check the
  package avoiding to get TIMEOUT warnings on the BioC build machines.

jorainer committed:
ci: force installation of faahKO package

jorainer committed:
ci: remove tab

jorainer committed:
Merge branch 'master' into jomaster

jorainer committed:
Merge pull request #494 from sneumann/jomaster

Adapt to MSnbase 2.15.3

jorainer committed:
bump version

jorainer committed:
Merge remote-tracking branch 'origin/master'

jorainer committed:
ci: change to BioC-check github action

jorainer committed:
ci: force install of the faahKO package

jorainer committed:
ci: install (github) xcms as a requirement for faahKO

jorainer committed:
ci: replace tabs with whitespaces

jorainer committed:
fix: fix C++ compiler WARNING in massifquant code

jorainer committed:
ci: force installation of faahKO also for Windows (and macOS?)

jorainer committed:
ci: update README to point to the new github action

jorainer committed:
feat: add `estimatePrecursorIntensity` function (issue #453)

- Add function `estimatePrecursorIntensity` to estimate the precursor intensity
  for MS2 spectra.
- Add related unit tests and documentation.
- Update LC-MS/MS vignette.

jorainer committed:
Merge branch 'master' into jomaster

jorainer committed:
ci: fix Bioc version

jorainer committed:
Merge pull request #496 from sneumann/jomaster

Add the `estimatePrecursorIntensity` function

jorainer committed:
docs: shorten examples

jorainer committed:
docs: remove link to xcmsPapply

jorainer committed:
feat: filter chrom peaks by intensity

- Add `FilterIntensityParam` for `refineChromPeaks` to filter chromatographic
  peaks based on their intensity (issue #502).
- Add related unit tests and documentation.

jorainer committed:
Merge pull request #504 from sneumann/jomaster

Filter chrom peaks by intensity

jorainer committed:
refactor: tiny changes to the internal .split_by_file function

jorainer committed:
refactor: improvements for per-file splitting

- Add and use `.split_per_file2` which splits `OnDiskMSnExp` and `XCMSnExp`
  objects more efficiently (but requires more memory), especially for very large
  data sets.

jorainer committed:
docs: disable parallel processing in all examples

jorainer committed:
Merge pull request #510 from sneumann/jomaster

Small updates and example performance improvements

jorainer committed:
ci: ensure RELEASE branches are tested against correct version

jorainer committed:
Merge remote-tracking branch 'origin/master' into jomaster

jorainer committed:
Re-add the deprecated parallel processing functions

- Re-add the deprecated parallel processing functions to fix issue
  https://github.com/sneumann/CAMERA/issues/63 in `CAMERA`.

jorainer committed:
Merge pull request #516 from sneumann/jomaster

Re-add old deprecated parallel processing functionality

nturaga committed:
bump x.y.z version to even y prior to creation of RELEASE_3_12 branch

nturaga committed:
bump x.y.z version to odd y following creation of RELEASE_3_12 branch

jorainer committed:
fix: support subsetting of XChromatograms with drop = FALSE

jorainer committed:
tests: ensure correct BioC version is used

jorainer committed:
Merge branch 'master' into jomaster

jorainer committed:
feat: add manualChromPeaks function (issue #525)

jorainer committed:
ci: disable windows and mac checks

jorainer committed:
ci: enable ubuntu

jorainer committed:
ci: replace GH with newest version from biocthis

jorainer committed:
ci: enable testthat, pkgdown etc in GH

jorainer committed:
ci: manually install additional packages

jorainer committed:
ci: disable BiocCheck

jorainer committed:
ci: install netcdf and smaller fixes

jorainer committed:
ci: disable installation of netcdf on macOS

jorainer committed:
Merge pull request #526 from sneumann/jomain

Add manual peak definitions as discussed in #525

jorainer committed:
feat: add filterIntensity function for Chromatograms

- Add `filterIntensity,Chromatogram` and `filterIntensity,MChromatograms`
  functions.
- Add parameter `method = "none"` to the `align,Chromatogram` function.

jorainer committed:
Merge pull request #527 from sneumann/jomain

feat: add filterIntensity function for Chromatograms

jorainer committed:
fix: smaller fixes and cleanup

- Chromatogram alignment uses `MsCoreUtils::closest` with `duplicates =
  "closest"`.
- Remove obsolete `.match_closest` function.

jorainer committed:
Merge pull request #529 from sneumann/jomain

fix: smaller fixes and cleanup

jorainer committed:
refactor: import Param class from ProtGenerics

- Remove the `Param` class from `xcms` and import from `ProtGenerics`.

jorainer committed:
refactor: move various methods to MSnbase

- Remove methods `align`, `normalize` and `filterIntensity` and import
  `alignRt`, `normalize` and `filterIntensity` for `Chromatogram` and
  `MChromatograms` objects from `MSnbase`.

jorainer committed:
ci: force installation of magick and faahKO

jorainer committed:
ci: force installation of devel versions

jorainer committed:
ci: use R devel on macOS

jorainer committed:
ci: install pre-build netcdf library for macos

jorainer committed:
ci: install also hdf5 for macOS

jorainer committed:
ci: fix new line

jorainer committed:
ci: update to netcdf 4.7.4

jorainer committed:
ci: add szip library for macOS

jorainer committed:
Merge pull request #530 from sneumann/param

Move various functionality to MSnbase

lgatto committed:
Depend on latest ProtGenerics

lgatto committed:
import quantify from ProtGenerics (previously from MSnbase)

jorainer committed:
Merge pull request #532 from lgatto/master

Import ProtGenerics::quantify (was MSnbase::quantify)

jorainer committed:
docs: add rmarkdown to Suggests

jorainer committed:
ci: force installation of rmarkdown

jorainer committed:
ci: force installation of BiocStyle

jorainer committed:
ci: try to fix failed install of dependencies

jorainer committed:
Update DESCRIPTION

jorainer committed:
ci: small fix

jorainer committed:
ci: update to newer biocthis version

jorainer committed:
ci: manually install faahKO package

jorainer committed:
ci: force installation of MSnbase from github

jorainer committed:
Merge pull request #535 from sneumann/jomain

Fix github actions

jorainer committed:
feat: add filterChromPeaks method

- Add `filterChromPeaks` method for `XChromatogram` and `XChromatograms`
  objects.

jorainer committed:
refactor: chromPeakSpectra returns Spectra and MS1 spectra

- `chromPeakSpectra` allows to return `Spectra` or `List` of `Spectra` (issue
  #539).
- `chromPeakSpectra` supports returning also of MS1 spectra.
- `chromPeakSpectra` supports returning the spectrum with the largest total
  signal or base peak intensity.

jorainer committed:
refactor: use .fdata instead of fData

jorainer committed:
docs: update LC-MS/MS vignette

jorainer committed:
ci: manual install of MsBackendMgf

jorainer committed:
refactor: featureSpectra support return of Spectra (issue #539)

- `featureSpectra` allows to return `Spectra` or `List` of `Spectra`.
- Add parameter `peaks` to `chromPeakSpectra` to support extracting spectra for
  selected peaks only.

jorainer committed:
feat: support to extract spectra for individual peaks or features

- Add support to extract spectra only for selected peaks or features in
  `chromPeakSpectra` and `featureSpectra`.

jorainer committed:
docs: drop momentarily the SWATH analysis from the LC-MS/MS vignette

jorainer committed:
refactor: remove all version info from classes

jorainer committed:
docs: add warning if processing queue is not empty

jorainer committed:
Merge branch 'no_version' into jomain

jorainer committed:
feat: allow return of Spectra in reconstructChromPeakSpectra

- Add parameter `return.type` to `reconstructChromPeakSpectra` (issue #539).

jorainer committed:
Update version requirements for MSnbase and ProtGenerics

jorainer committed:
ci: disable forced install of vignettes

jorainer committed:
Merge pull request #540 from sneumann/jomain

Updates

jorainer committed:
Export findmzROI method

jorainer committed:
docs: add missing documentation for findmzROI

jorainer committed:
Merge pull request #543 from sneumann/jomain

Export findmzROI method

jorainer committed:
feat: add filterChromPeaks,XCMSnExp (issue #541)

- Add the `filterChromPeaks,XCMSnExp` method and update documentation for filter
  functions.

jorainer committed:
Merge pull request #544 from sneumann/jomain

feat: add filterChromPeaks,XCMSnExp (issue #541)

sneumann committed:
Bumped some minimal dependency version

jorainer committed:
fix: issue #545

jorainer committed:
docs: remove LC-MS/MS vignette and drop MsBackendMgf dependency

jorainer committed:
Merge pull request #547 from sneumann/jomain

Temporarily remove the LC-MS/MS vignette

jorainer committed:
feat: add featureArea function

- Add the `featureArea` function that returns the (median) m/z and retention
  time range of all chromatraphic peaks associated with a feature.

jorainer committed:
fix: fix featureSpectra

- Reimplement `featureSpectra`.

jorainer committed:
docs: re-add LC-MS/MS analysis vignette

jorainer committed:
ci: force installation of MsBackendMassbank

jorainer committed:
ci: fix spelling mistake in github repo name

jorainer committed:
ci: remove MsBackendMgf from remotes and fix dependency

jorainer committed:
Merge pull request #554 from sneumann/jomain

Re-add the LC-MS/MS vignette and fix featureSpectra

sneumann committed:
Feature: plotQC() supports XCMSnExp objects now

sneumann committed:
Merge branch 'master' into feature/plotQCxcms2

jorainer committed:
Merge pull request #555 from sneumann/feature/plotQCxcms2

Feature/plot q cxcms2

jorainer committed:
refactor: use inherits instead of class

sneumann committed:
Fix plotQC() for XCMSnExp objects

nturaga committed:
bump x.y.z version to even y prior to creation of RELEASE_3_13 branch

nturaga committed:
bump x.y.z version to odd y following creation of RELEASE_3_13 branch

jorainer committed:
Merge branch 'master' into jomain

rickhelmus committed:
SSgauss: update for R 4.1

sneumann committed:
Bump version after merging #562

Rick informed us that: nls() now passes the trace and control arguments to getInitial(),
notably for all self-starting models, so these can also be fit in zero-noise situations via a scaleOffset.
For this reason, the initial function of a selfStart model must now have ... in its argument list.

sneumann committed:
Merge branch 'rickhelmus-master'

sneumann committed:
Bump BioC test environment to 3.15

sneumann committed:
Revert "Bump BioC test environment to 3.15"

This reverts commit 5c4cae7330ef5d74883d056636acf9e96c22f097
which was obviously the wrong approach.

jorainer committed:
Merge branch 'master' into jomain

jorainer committed:
fix: fix LC-MS/MS vignette and unit tests (issue #564)

jorainer committed:
fix: replace |> with %>%

sneumann committed:
Merge branch 'jomain'

jorainer committed:
feat: add groupFeatures and first feature grouping approaches

- Add first feature grouping functionality (issue #566).

jorainer committed:
fix: small issue

jorainer committed:
refactor: move chromatogram correlation functionality to MSnbase

jorainer committed:
refactor: re-use compareChromatograms from MSnbase

jorainer committed:
feat: add feature grouping based on EIC similarity

jorainer committed:
feat: add preprocessed data

jorainer committed:
feat: add plotChromatogramsOverlay

- Add `plotChromatogramsOverlay` function.
- Finalize addition of `groupFeatures` functionality (issues #566 and #567).

jorainer committed:
fix: chromPeaks: ensure rt and mz are converted to numbers

jorainer committed:
fix: adapt methods to generics in ProtGenerics

jorainer committed:
fix: change package for vignette

jorainer committed:
ci: install also magick

jorainer committed:
ci: disable most vignette code blocks

jorainer committed:
ci: use BioC devel

jorainer committed:
docs: fix package in vignette

jorainer committed:
fix: CI, restore vignette and drop import of MSnbase::bin

jorainer committed:
ci: fix config file

jorainer committed:
ci: force installation of mzR from github

jorainer committed:
docs: disable most of the vignette

jorainer committed:
docs: fix package in vignette

jorainer committed:
docs: restore full vignette

jorainer committed:
refactor: add parameter transform to plotChromatogramsOverlay

- Add parameter `transform` to `plotChromatogramsOverlay`.

sneumann committed:
Update plotQC.R

jorainer committed:
feat: add manualFeatures function

- Add `manualFeatures` which allows to manually define and add features to an
  `XCMSnExp` object.

jorainer committed:
docs: update documentation to introduce "compounding"

- Update the documentation to introduce the term "compounding" for feature
  grouping.

sneumann committed:
Merge pull request #574 from sneumann/fix/mtbls2

Update plotQC.R

jorainer committed:
docs: expand documentation on featureDefinitions

jorainer committed:
Merge pull request #572 from sneumann/ms_features

Add LC-MS feature grouping functionality to xcms

jorainer committed:
fix: plot type XIC with empty data

jorainer committed:
refactor: skip repeated peak indexing for features if not needed

- Avoid re-indexing of peak-to-feature assignments if not necessary. This leads
  to performance improvements specifically for MS1-only data.

jorainer committed:
docs: small update

jorainer committed:
fix: remove warning check from unit test

sneumann committed:
Merge pull request #581 from sneumann/jomain

Small updates and unit test fix

sneumann committed:
Disable testing on windows i386 and Disable parallel processing on Windows

sneumann committed:
Merge pull request #583 from sneumann/fix/issue582

Disable testing on windows i386 and Disable parallel processing on Windows

nturaga committed:
bump x.y.z version to odd y following creation of RELEASE_3_14 branch

nturaga committed:
bump x.y.z version to even y prior to creation of RELEASE_3_14 branch

jorainer committed:
fix: avoid error message from issue #584

- Add ... to findChromPeaks,OnDiskMSnExp to avoid error message from issue #584.

jorainer committed:
fix: fix bug in EIC feature grouping

- Fix generation of non-symmetrix similarity matrix in EIC-based feature
  grouping.

jorainer committed:
fix: fix unit test

sneumann committed:
Merge branch 'jomain'

sneumann committed:
Use mzML instead of mzData files in testing and vignettes

sneumann committed:
Surviving mzR without mzData

sneumann committed:
Merge pull request #586 from sneumann/fix/removeMzDataTests

Fix/remove mz data tests

sneumann committed:
Bump version

jorainer committed:
feat: add `transformIntensity` method

jorainer committed:
docs: fix missing documentation

jorainer committed:
refactor: use intensity weighted mean to calculate m/z for scan

- Use an intensity weighted mean instead of a simple mean to report the m/z for
  a scan (spectrum) in EIC reporting with the centWave peak detection
  algorithm (issue #590).

jorainer committed:
fix: fix issue 603

sneumann committed:
Merge pull request #604 from sneumann/jomain

Closes #604 and adds intensity transformation

jorainer committed:
Merge branch 'master' into fix_peak_mz

jorainer committed:
docs: increase version

jorainer committed:
refactor: allow to choose m/z calculation function

zeehio committed:
Replace links to MassSpecWavelet::sav.gol() with links to MassSpecWavelet::peakDetectionCWT().

See https://github.com/zeehio/MassSpecWavelet/issues/2

jorainer committed:
feat: add parameter expandMz to featureChromatograms

- Add parameter `expandMz` to `featureChromatorgrams` (#612).
- Add package `signal` to Suggests.

jorainer committed:
refactor: rewrite function to subset features and chrom peaks

sneumann committed:
Merge pull request #606 from sneumann/fix_peak_mz

Fix peak mz

sneumann committed:
Revert "refactor: allow to choose m/z calculation function"

This reverts commit e99ee747da84fb4a5df077d45587db31b7ed7bb5.
The functionality was only for testing implementation differences,
and documented in https://github.com/sneumann/xcms/pull/606/

sneumann committed:
fix merge conflict

sneumann committed:
Merge branch 'jomain'

lshep committed:
fix NEWS formatting to properly display

nturaga committed:
bump x.y.z version to even y prior to creation of RELEASE_3_15 branch

nturaga committed:
bump x.y.z version to odd y following creation of RELEASE_3_15 branch

jorainer committed:
ci: update to BioC 3.16

jorainer committed:
ci: install dependencies from BioC

jorainer committed:
ci: change to git global command

jorainer committed:
refactor: use rbindFill instead of rbind in groupChromPeaks

jorainer committed:
Merge pull request #620 from zeehio/relink-savgol

[DOC] Replace links to MassSpecWavelet::sav.gol() with links to peakDetectionCWT()

jorainer committed:
Merge remote-tracking branch 'origin/master'

jorainer committed:
doc: update documentation

sneumann committed:
Fix typo in list element lookup, closes #623

jorainer committed:
docs: update/expand documentation for firstBaselineCheck parameter

jwokaty committed:
bump x.y.z version to odd y following creation of RELEASE_3_16 branch

jwokaty committed:
bump x.y.z version to even y prior to creation of RELEASE_3_16 branch

jorainer committed:
Merge remote-tracking branch 'upstream/master'

jorainer committed:
ci: use BioC 3.17

jorainer committed:
Merge remote-tracking branch 'origin/master' into jomain

jorainer committed:
fix: issue #654

- Fix problem with `fillChromPeaks` after `MatchedFilterParam` peak detection on
  very sparse data (i.e. data with many empty spectra).

dmurdoch committed:
Change rgl.* to *3d

jorainer committed:
docs: fix documentation

jorainer committed:
Merge pull request #655 from sneumann/jomain

Fix problem in fillChromPeaks with sparse data

jorainer committed:
Merge pull request #654 from dmurdoch/rgl

Change rgl.* to *3d

jorainer committed:
Update NEWS

jorainer committed:
refactor: improve reconstructChromPeakSpectra

- Reimplement the `reconstructChromPeakSpectra` function for DIA data analysis
  to fix an issue with chrom peaks in overlapping isolation windows for SWATH
  data and to generally improve performance.

jorainer committed:
refactor: clean up and remove old functions

jorainer committed:
docs: re-add author information

sneumann committed:
Merge pull request #662 from sneumann/jomain

Improve performance for SWATH/DIA data analysis

sneumann committed:
Several fixes to the longtests

stanstrup commented:
shouldn't this be closed?
